src/sas/qtgui/Utilities/FileConverter.py

```diff
@@ -0,0 +1,487 @@
+        """
+        file_candidate = self.openFile()
+        if not file_candidate:
+            return
+        self.ifile = file_candidate
+        self.txtIFile.setText(os.path.basename(str(file_candidate)))
+        self.updateConvertState()
+
+    def onQFileOpen(self):
+        """
+        Show the path chooser for file with Q
+        """
+        file_candidate = self.openFile()
+        if not file_candidate:
+            return
+        self.qfile = file_candidate
+        self.txtQFile.setText(os.path.basename(str(file_candidate)))
+        self.updateConvertState()
+
+    def openFile(self):
+        """
+        Show the path chooser for existent file
+        """
+        datafile = None
+        try:
+            datafile = QtWidgets.QFileDialog.getOpenFileName(
+                self, "Choose a file", "", "All files (*.*)")[0]
+        except (RuntimeError, IOError) as ex:
+            log_msg = "File Converter failed with: {}".format(ex)
+            logging.error(log_msg)
+            raise
+        return datafile
+
+    def getDetectorMetadata(self):
+        """
+        Read the detector metadata fields and put them in the dictionary
+        """
+        detector = Detector()
+        detector.name = self.txtMD_Name.text()
+        detector.distance = Utilities.toFloat(self.txtMD_Distance.text())
+        detector.offset = Vector(x=Utilities.toFloat(self.txtMD_OffsetX.text()),
+                                 y=Utilities.toFloat(self.txtMD_OffsetY.text()))
+        detector.orientation = Vector(x=Utilities.toFloat(self.txtMD_OrientRoll.text()),
+                                      y=Utilities.toFloat(self.txtMD_OrientPitch.text()),
+                                      z=Utilities.toFloat(self.txtMD_OrientYaw.text()))
+        detector.beam_center = Vector(x=Utilities.toFloat(self.txtMD_BeamX.text()),
+                                      y=Utilities.toFloat(self.txtMD_BeamY.text()))
+        detector.pixel_size = Vector(x=Utilities.toFloat(self.txtMD_PixelX.text()),
+                                     y=Utilities.toFloat(self.txtMD_PixelY.text()))
+        detector.slit_length = Utilities.toFloat(self.txtMD_Distance.text())
+        return detector
+
+    def getSampleMetadata(self):
+        """
+        Read the sample metadata fields and put them in the dictionary
+        """
+        sample = Sample()
+        sample.name = self.txtMSa_Name.text()
+        sample.beam_size = Vector(x=Utilities.toFloat(self.txtMSo_BeamSizeX.text()),
+                                  y=Utilities.toFloat(self.txtMSo_BeamSizeY.text()))
+        sample.beam_shape = self.txtMSo_BeamShape.text()
+        sample.wavelength = Utilities.toFloat(self.txtMSo_BeamWavelength.text())
+        sample.wavelength_min = Utilities.toFloat(self.txtMSo_MinWavelength.text())
+        sample.wavelength_max = Utilities.toFloat(self.txtMSo_MaxWavelength.text())
+        sample.wavelength_spread = Utilities.toFloat(self.txtMSo_Spread.text())
```

Comment:
Shouldn't all of this be in getSourceMetadata()?

Reply:
yes, it should. fixed